resource-service.js

```diff
@@ -187,7 +187,8 @@ function factory($rootScope, $http, $location, brModelService) {
     self.startLoading();
     var config = self._buildConfig(options);
     var url = options.url || resource.id;
-    return Promise.resolve($http.post(url, resource, config))
+    var updateMethod = options.method === 'patch' ? $http.patch : $http.post;
+    return Promise.resolve(updateMethod(url, resource, config))
```

Comment:
Could perhaps make this more generic to handle any $http method:
```js
return Promise.resolve($http[options.method || 'post'](url, resource, config))
```
This would expose $http implementation details which is maybe a bad idea.